include/cuco/detail/static_set/static_set_ref.inl

```diff
@@ -343,6 +343,144 @@ class operator_impl<op::contains_tag,
   }
 };
 
+template <typename Key,
+          cuda::thread_scope Scope,
+          typename KeyEqual,
+          typename ProbingScheme,
+          typename StorageRef,
+          typename... Operators>
+class operator_impl<op::find_tag,
+                    static_set_ref<Key, Scope, KeyEqual, ProbingScheme, StorageRef, Operators...>> {
+  using base_type  = static_set_ref<Key, Scope, KeyEqual, ProbingScheme, StorageRef>;
+  using ref_type   = static_set_ref<Key, Scope, KeyEqual, ProbingScheme, StorageRef, Operators...>;
+  using key_type   = typename base_type::key_type;
+  using value_type = typename base_type::value_type;
+  using iterator   = typename StorageRef::iterator;
+  using const_iterator = typename StorageRef::const_iterator;
+
+  static constexpr auto cg_size     = base_type::cg_size;
+  static constexpr auto window_size = base_type::window_size;
+
+ public:
+  /**
+   * @brief Returns a const_iterator to one past the last slot.
+   *
+   * @note This API is available only when `find_tag` is present.
+   *
+   * @return A const_iterator to one past the last slot
+   */
+  [[nodiscard]] __host__ __device__ constexpr const_iterator end() const noexcept
+  {
+    auto const& ref_ = static_cast<ref_type const&>(*this);
+    return ref_.storage_ref_.end();
+  }
+
+  /**
+   * @brief Returns an iterator to one past the last slot.
+   *
+   * @note This API is available only when `find_tag` is present.
+   *
+   * @return An iterator to one past the last slot
+   */
+  [[nodiscard]] __host__ __device__ constexpr iterator end() noexcept
+  {
+    auto const& ref_ = static_cast<ref_type const&>(*this);
+    return ref_.storage_ref_.end();
+  }
```

Comment:
I've thought about this. We currently expose `iterator` and `end()` only when `find_tag` is present since:

1. other functionalities don't really use `iterator`
2. The `iterator` we defined is more of a strong type of slot pointer other than an iterator since it's not incrementable thus cannot be used to actually "iterate" the underlying data structure. Exposing it to the open public may cause misuse

I'm not saying no to your suggestions but just share the thoughts I have in mind for this design.

Reply:
Two very valid points. My only concern would be that there might be other operators that return such an iterator type. I we encounter this scenario in the future, we can still move the definition to a different location. Let's leave it as is for this PR.

Reply:
Another use case would be `insert_and_find` where the return value is pair of an iterator and a boolean indicating whether the insertion was successful or not.